### PR TITLE
blog: March '22 Heartbeat

### PR DESCRIPTION
**Pull request by jendefig**
Status:  Ready for review @flippedcoder!  Links are being skipped, so I still need to check manually.

- [x]  There are no grammatical errors, jargon, or obvious stylistic issues 
- [x]  Have images been run through tiny.png?
- [x]  Are all lines 80 characters or less?
- [x]  Are Oxford commas observed?
- [x]  Content Creator has done a complete review for typos/language after taking a break from content
- [x]  links checked in piece
- [x]  Have reviewed what the piece looks like on your own [[machine using yarn develop](https://dvc.org/doc/user-guide/contributing/docs#development-environment)](https://dvc.org/doc/user-guide/contributing/docs#development-environment)

### CODE REVIEW

**Comment by jendefig** [comment]
@rogermparent Having link issues with Towards Data Science links.

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+descriptionLong: |
+  This month you will find:
+
+    🧘🏻‍♀️ MLOps is a mess, but that's ok,
+    
+    🥰 Tutorials and workflows from the Community,
+
+    🗣 Upcoming Events,
+
+    🔺 MLOps Maturity Models
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:18]: Should a comma be here like with the other topics?

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+## Mihail Eric: MLOps is a Mess But That's to be Expected
+
+[**Mihail Eric**](https://twitter.com/mihail_eric) writes a long, but _really
+worth it_ piece entitled
+[MLOps is a Mess But That’s to be Expected.](https://www.mihaileric.com/posts/mlops-is-a-mess/)
+In it he discusses the allure of seeking a machine learning career, only run
+smack into the giant wall of learning that encompasses the space, not the least
+of which is the multitude of tools to pick through once you get there. The state
+of machine learning is reviewed, some history of DevOps for perspective on MLOps
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:49]:
```suggestion
of machine learning is reviewed and some history of DevOps for perspective on MLOps
```

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+## Matthew Upson: MLOps for Conversational AI with Rasa, DVC, and CML (PartII)
+
+In the [December Heartbeat,](https://dvc.org/blog/december-21-heartbeat) I told
+you about [**Matt Upson's**](https://twitter.com/m_a_upson) first post in his
+series on using DVC, CML and Rasa together.
+[In this second post](https://medium.com/mantisnlp/mlops-for-conversational-ai-with-rasa-dvc-and-cml-part-ii-3a70fe2f357d)
+he goes through some Rasa basics and gets the DVC pipeline setup, with its train
+and test stages, params, dependencies, outs and metrics. He also covers syncing
+with DVC, making changes, the `dvc repro` command, the ./dvc lock file, and
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:96]:
```suggestion
with DVC, making changes, the `dvc repro` command, the `.dvc-lock` file, and
```

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+metrics diff in Rasa project
+([Source link](https://medium.com/mantisnlp/mlops-for-conversational-ai-with-rasa-dvc-and-cml-part-ii-3a70fe2f357d))_
+
+## Sibanjan Das: MLOps for Enterprise AI
+
+[**Sibanjan Das**](https://www.linkedin.com/in/sibanjan/) notes the trending of
+the MLOps keyword in
+[his piece](https://dzone.com/articles/mlops-for-enterprise-ai) in
+[DZone.](https://dzone.com) Sibanjan gives an overview of MLOps and how it
+supports the AIML ecosystem to deliver return on investment for ML projects. He
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:110]:
```suggestion
supports the AI/ML ecosystem to deliver return on investment for ML projects. He
```

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+an overview of the use of Tensor Flow, PyTorch, and Docker, as well as DVC for
+version control, experiment management deployment, and collaboration. Additional
+resources from Xenonstack are provided for further review.
+
+### Yuqi Li: Why MLOps should be Open Source
+
+![Why MLOps Tools should be Open Source](/uploads/images/2022-03-16/yuqi-li.jpeg 'Why MLOps Tools should be Open Source :wrap-right ==300')
+[**Yuqi Li**](https://www.linkedin.com/in/yuqiliofficial/)
+[in this opinion piece,](https://towardsdatascience.com/why-mlops-tools-should-be-open-source-5ad696463f54)
+In [Towards Data Science.](https://towardsdatascience.com/) overviews the
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:141]:
```suggestion
in [Towards Data Science.](https://towardsdatascience.com/) overviews the
```

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+
+![Community Driven Learning](/uploads/images/2022-03-16/community.jpeg '=800')
+_Community Driven Learning
+([Source link: Unsplash by john_cameron](https://unsplash.com/@john_cameron))_
+
+## And speaking of learning…
+
+# Company News
+
+https://giphy.com/clips/flying-jetpack-jet-pack-AFInqMxZCWfEiedpBs
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:174]: Should this be displaying the gif?

File: content/blog/2022-03-16-march-22-heartbeat.md (hunk 1 above)
Comment (jendefig) [reply content/blog/2022-03-16-march-22-heartbeat.md:18]: Yep, thanks!

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+- We have already begun working on the second course which will be more advanced
+  (remember those maturity models outlined in the article from DZone above?) and
+  will cover scenarios with CML. We are also working on creating an ebook for
+  each video that will provide relevant information, diagrams, and links with
+  the video content instead of being batched at the end of the module. The ebook
+  format will also let you take your own notes as you study!
+
+## New Hires
+
+https://giphy.com/clips/flying-jetpack-jet-pack-AFInqMxZCWfEiedpBs
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:196]: Should this be displaying the gif too?

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+
+[**Mike Moynihan**](https://www.linkedin.com/in/michael-moynihan/) joins us from
+Brookly, NY as an Account Executive. He previously worked at Code Climate as the
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:199]:
```suggestion
Brooklyn, NY as an Account Executive. He previously worked at Code Climate as the
```

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+
+## Upcoming Events
+
+### March Office Hours!
+
+Be sure to join us at the
+[March Office Hours Meetup,](https://www.meetup.com/Machine-Learning-Engineer-Community-Virtual-Meetups/events/283998696/)
+where [**Fabian Zills**,](https://github.com/PythonFZ/) PhD student at
+[University of Stuttgart,](https://www.uni-stuttgart.de/en/) will present his
+ZnTrack ("zinc track") project which creates, runs and benchmarks DVC pipelines
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:223]: Could we add a link to the ZnTrack GitHub repo?

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+href="https://www.meetup.com/Machine-Learning-Engineer-Community-Virtual-Meetups/events/283998696/"
+title="March Office Hours - ZnTrack"
+description="RSVP for DVC Office Hours - ZnTrack - Create, Visualize, Run and Benchmark DVC Pipelines in Python & Jupyter Notebooks "
+link="https://meetup.com"
+image="/uploads/images/2022-03-16/office-hours-meetup.png"/>
+
+## Conferences/Hackathons
+
+- We will be sponsoring [ODSC East](https://odsc.com/boston/) and
+  [MLOPs World](https://mlopsworld.com/) this year, so if you are attending,
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:236]:
```suggestion
  [MLOps World](https://mlopsworld.com/) this year, so if you are attending,
```

File: content/blog/2022-03-16-march-22-heartbeat.md [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+- [**David de la Iglesia Castro**](https://github.com/daavoo) will be presenting
+  his workshop "Making MLOps Uncool Again" at
+  [MLOps World New York](https://mlopsworld.com/newyork/) on March 29th and at
+  [PyCon Berlin](https://2022.pycon.de/) April 11th.
+- Community member [**Gift Ojeabulu**](https://twitter.com/GiftOjeabulu_) will
+  be giving a talk on "MLops Exploration with Git and DVC for Machine Learning
+  Project" at [Open Source Festival 2022](https://festival.oscafrica.org/) March
+  24-26
+- [BatteryDev Hackathon](https://www.battery.dev/) will take place next week and
+  [**Milecia McGregor**](https://twitter.com/FlippedCoding)will hold an Office
```
Comment (flippedcoder) [inline content/blog/2022-03-16-march-22-heartbeat.md:250]:
```suggestion
  [**Milecia McGregor**](https://twitter.com/FlippedCoding) will hold an Office
```

**Comment by jendefig** [comment]
This is ready to go except for the links that I don't know what to do about.  If it can't go today, I need to change the dates on everything and delete a section (tomorrow's meetup). @shcheklein @jorgeorpinel @rogermparent

**Review by shcheklein — APPROVED** [review]

**Comment by shcheklein** [comment]
@jendefig LGTM ... I've approved it, feel free to Squash and merge when you are ready.

**Comment by jendefig** [comment]
@shcheklein has approved, but hold. I may add a statement for the Community about Ukraine.

**Comment by rogermparent** [comment]
I investigated the towardsdatascience links and had a rough time pinpointing what's wrong, but I think I found it: TDS returns 503 when given `node-fetch`'s default user agent.

How I diagnosed it:

- link-check CLI fails with the same code 503
- the links are fine in a browser
- default curl returns 200 (OK)
- Lychee (a well established link checking library) succeeds when given a towardsdatascience link

After getting a hunch that something in our request was the problem, I snooped into the requests link-check was sending by running the link check CLI with `NODE_DEBUG=http`. Nothing else looked off, so I tried using our link-check's user agent on curl by running the command `curl -A 'node-fetch/1.0 (+https://github.com/bitinn/node-fetch)' -L --head -X HEAD https://towardsdatascience.com/`, which finally returned a `HTTP/2 503`- the failing code we're looking for.

Pinging @julieg18 since this info should be helpful where she's looking into our link checker right now. I should be able to get together an interim solution if we need it, but weird problems like this are a big reason why I'd personally throw my hat behind using a popular battle-tested library like `lychee` for our planned link-check revamp.

**Comment by jendefig** [comment]
@julieg18 @rogermparent Need some help here.  

I just put in a message on Ukraine at the top.  I want to include the headings and bulleted links inside the details.  Is that possible?  I had them all aligned left and that didn't work and then I tabbed it all in, but that didn't work either.

Alternatively I could make a new notion page with all those links and make it public and just put that link in there.  LMK.

@shcheklein @jorgeorpinel Could use review of the message please. 🙏

**Comment by jendefig** [comment]
> I just put in a message on Ukraine at the top. I want to include the headings and bulleted links inside the details. Is that possible? I had them all aligned left and that didn't work and then I tabbed it all in, but that didn't work either.

Nevermind!  @julieg18 @rogermparent Apparently tabbing them in did work! 🎉

**Comment by jendefig** [comment]
This has been approved by @shcheklein.  There are some new links that have the same problem as Toward Data Science ones, but they work in the deployment.  Can someone please merge this for me? @julieg18 @rogermparent?

**Pull request merged**